crates/fuel-gas-price-algorithm/src/v1.rs

```diff
@@ -243,11 +236,44 @@ impl AlgorithmUpdaterV1 {
         }
     }
 
+    fn update_rewards(&mut self, fee_wei: u64) {
+        let block_da_reward = self.da_portion_of_fee(fee_wei);
+        self.total_da_rewards_excess = self
+            .total_da_rewards_excess
+            .saturating_add(block_da_reward.into());
+    }
+
+    fn update_projected_cost(&mut self, block_bytes: u64) {
+        let block_projected_da_cost =
+            (block_bytes as u128).saturating_mul(self.latest_da_cost_per_byte);
+        self.projected_total_da_cost = self
+            .projected_total_da_cost
+            .saturating_add(block_projected_da_cost);
+    }
+
+    // Take the `fee_wei` and return the portion of the fee that should be used for paying DA costs
+    fn da_portion_of_fee(&self, fee_wei: u64) -> u64 {
+        // fee_wei * (da_price / (exec_price + da_price))
+        let numerator = fee_wei.saturating_mul(self.descaled_da_price());
+        let denominator = self
+            .descaled_exec_price()
+            .saturating_add(self.descaled_da_price());
+        if denominator == 0 {
+            0
+        } else {
+            numerator.div_ceil(denominator)
+        }
+    }
+
     // We are assuming that the difference between u128::MAX and i128::MAX is negligible
     fn projected_cost_as_i128(&self) -> i128 {
         i128::try_from(self.projected_total_da_cost).unwrap_or(i128::MAX)
     }
 
+    fn rewards_as_i128(&self) -> i128 {
+        i128::try_from(self.total_da_rewards_excess).unwrap_or(i128::MAX)
```

Comment:
qq: is there ever a chance that the `total_da_rewards_excess` will not be converted successfully?

Reply:
It just depends on what we define as "successful". Essentially, the `i128` is way way bigger than we'll ever need. 

Above I have a comment that isn't accurate:
```rs
     // We are assuming that the difference between u128::MAX and i128::MAX is negligible
```
in reality
```
u128::MAX: 340282366920938463463374607431768211455
i128::MAX:     170141183460469231731687303715884105727
```

But I don't know that any other behavior is better. i.e. is there a more explicit way we could define these types to allow the arithmetic but also capture the intended range. I'm not sure.

Maybe we could rename this `clamped_rewards_as_i128` or something?

Reply:
I went with that for now.